### PR TITLE
add API for best practices

### PR DESCRIPTION
**Pull request by Azdaroth**

### CODE REVIEW

File: content/guides/api-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+* Instead of querying API every few minutes, take advantage of webhooks and get notified when some record is created, updated or deleted. However, the order of the webhooks' delivery is not guaranteed (especially important when there are multiple updates in a short period), so we suggest to schedule a synchronization when the webhook is delivered, instead of trusting that the data from the last webhook represents the most recent projection of a given record. Nevertheless, if you use strategies similar to Event Sourcing and eventual consistency is good enough for your application, you can apply the events based on their timestamps and have the objects eventually consistent.
+
+* Webhook delivery might fail for various reasons (e.g., network problems, application servers being not reachable), it is still recommended to periodically synchronize data, even if you subscribe to all possible events your application needs.
+
+* Using Ruby on Rails? Don't reinvent the wheel! Take advantage of multiple gems we use for our internal applications:
+
+1.  [**bookingsync-api**](https://github.com/BookingSync/bookingsync-api) - API client
```
Comment (JulienItard) [inline content/guides/api-best-practices.md]: indent

File: content/reference/changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## master
+  * [api improvement] Add guides for API best practices
```
Comment (JulienItard) [inline content/reference/changelog.md]: You added only one guide so no S

File: content/reference/changelog.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## master
```
Comment (StoneFrog) [inline content/reference/changelog.md]: not needed

**Comment by Azdaroth** [comment]
@StoneFrog Done.

**Pull request merged**